app/endpoints/users.py

```diff
@@ -579,6 +568,102 @@ async def reset_password(
     return standard_responses.Result()
 
 
+@router.post(
+    "/users/migrate-mail",
+    status_code=204,
+    tags=[Tags.users],
+)
+async def migrate_mail(
+    mail_migration: schemas_core.MailMigrationRequest,
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member),
+    settings: Settings = Depends(get_settings),
+):
+    """
+    Due to a change in the email format, all student users need to migrate their email address.
+    This endpoint will send a confirmation code to the user's new email address. He will need to use this code to confirm the change with `/users/confirm-mail-migration` endpoint.
+    """
+
+    if not re.match(
+        r"^[\w\-.]*@((ecl\d{2})|(alternance\d{4})|(master)).ec-lyon.fr$", user.email
+    ):
+        raise HTTPException(
+            status_code=400,
+            detail="Only student users with an old email adresse can migrate their email address",
+        )
+
+    if not re.match(r"^[\w\-.]*@etu(-enise)?.ec-lyon.fr$", mail_migration.new_email):
+        raise HTTPException(
+            status_code=400,
+            detail="The new email adresse must match the new ECL format for student users",
+        )
+
+    confirmation_token = security.generate_token()
+
+    migration_object = models_core.CoreUserEmailMigrationCode(
+        user_id=user.id,
+        new_email=mail_migration.new_email,
+        confirmation_token=confirmation_token,
+    )
+
+    await cruds_users.create_email_migration_code(
+        migration_object=migration_object, db=db
+    )
+
+    if settings.SMTP_ACTIVE:
+        send_email(
+            recipient=mail_migration.new_email,
+            subject="MyECL - Confirm your new email adresse",
+            content=f"You can confirm your new email adresse by clicking the following link: {settings.CLIENT_URL}/users/migrate-mail-confirm?token={confirmation_token}",
+            settings=settings,
+        )
+    print(
```

Comment:
Is this print necessary ?

Reply:
The print make it easier to access the token when working with a local environment without SMTP. I wouldn't mind removing it, but I think we should then search an alternative

Reply:
We don't have a reliable way to know if Hyperion is in production but we could maybe only print the token when email are disabled

Reply:
If it is really hard to implement, better don't waste time, on it, it will not flood the server anyway